components/dropdown/dropdown-menu.js

```diff
@@ -12,20 +12,29 @@ import { ThemeMixin } from '../../mixins/theme-mixin.js';
  */
 class DropdownMenu extends ThemeMixin(DropdownContentMixin(LitElement)) {
 
+	static get properties() {
+		return {
+			_noAutoFocus: { type: Boolean }
+		};
+	}
+
 	static get styles() {
 		return dropdownContentStyles;
 	}
 
 	constructor() {
 		super();
-		this.noAutoFocus = true;
 		this.noPadding = true;
 		this._maxHeightNonTray = this.maxHeight;
+		this._noAutoFocus = false;
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
+		if (this.noAutoFocus) this._noAutoFocus = true;
+		this.noAutoFocus = true;
```

Comment:
A bit confused about this line as we will set noAutoFocus to be true for the dropdownContentMixin as well. What if some menus still want auto-focus? Is that why you added the check to line 126?

Reply:
Yup so how this currently works (before this change) is that `this.noAutoFocus` was always set to true in the constructor so that `DropdownContentMixin`'s auto focus logic wouldn't be used (https://github.com/BrightspaceUI/core/blob/main/components/dropdown/dropdown-content-mixin.js#L554) and it would then do its own thing where it would use `menu.focus()` (https://github.com/BrightspaceUI/core/blob/main/components/dropdown/dropdown-menu.js#L117). This meant that setting the `no-auto-focus` attribute on `d2l-dropdown-menu` would never be respected, which is why I made this change. Then in line 126 it will still use `menu.focus()` as before, unless `no-auto-focus` was used.